src/metabase/api/setting.clj

```diff
@@ -5,33 +5,51 @@
             [metabase.models.setting :as setting]
             [metabase.util.schema :as su]))
 
+(defn- do-with-setting-access-control
+  [thunk]
+  (try
+    (binding [setting/*enforce-setting-access-checks* true]
+      (thunk))
+    (catch clojure.lang.ExceptionInfo e
+      ;; Throw a generic 403 for non-admins, so as to not reveal details about settings
```

Comment:
Do we want to throw the whole exception even for admins?

Reply:
That's the status quo, so I figured I wouldn't change it. If an admin gets an exception here it won't be a permission issue, it'll be something like trying to write a read-only setting or a misc server error.